internal/dcache/file_manager/utils.go

```diff
@@ -63,27 +67,67 @@ func isOffsetChunkStarting(offset int64, fileLayout *dcache.FileLayout) bool {
 	return (offset%fileLayout.ChunkSize == 0)
 }
 
+func getMVForChunk(chunk *StagedChunk, fileLayout *dcache.FileLayout) string {
+	numMvs := int64(len(fileLayout.MVList))
+	return fileLayout.MVList[chunk.Idx%numMvs]
+}
+
 // Does all file Init Process for creation of the file.
 func NewDcacheFile(fileName string) (*DcacheFile, error) {
 	fileMetadata := &dcache.FileMetadata{
 		Filename: fileName,
+		State:    dcache.Writing,
 	}
-	// todo : assign uuid for fileID
-	// todo : get chunkSize and Stripe Size from the config and assign.
-	// todo : Choose appropriate MV's from the online mv's returned by the clustermap
+	fileMetadata.FileID = gouuid.New().String()
+	common.Assert(common.IsValidUUID(fileMetadata.FileID))
+
+	chunkSize := clustermap.GetCacheConfig().ChunkSize
+	stripeSize := clustermap.GetCacheConfig().StripeSize
+
+	common.Assert(stripeSize%chunkSize == 0, fmt.Sprintf("Stripe Size %d is not divisibe by chunkSize %d", stripeSize, chunkSize))
+	numMVs := stripeSize / chunkSize
+
 	fileMetadata.FileLayout = dcache.FileLayout{
-		ChunkSize:  4 * 1024 * 1024,
-		StripeSize: 16 * 1024 * 1024,
-		MVList:     []string{"mv0", "mv1", "mv2"},
+		ChunkSize:  int64(chunkSize),
+		StripeSize: int64(stripeSize),
+		MVList:     make([]string, numMVs),
 	}
+
+	// Get active MV's from the clustermap
+	activeMvs := clustermap.GetActiveMVs()
+	common.Assert(len(activeMvs) >= int(numMVs))
+	// todo : is there any better policy to pick?
+	i := 0
+	for k, _ := range activeMvs {
+		fileMetadata.FileLayout.MVList[i] = k
+		i++
+		if i == int(numMVs) {
+			break
+		}
+	}
+
+	log.Debug("DistributedCache[FM]::NewDcacheFile : Choose MV's are %s, %s, %s,  file: %s",
+		fileMetadata.FileLayout.MVList[0], fileMetadata.FileLayout.MVList[1], fileMetadata.FileLayout.MVList[2], fileName)
+
 	fileMetadataBytes, err := json.Marshal(fileMetadata)
 	if err != nil {
 		log.Err("DistributedCache[FM]::NewDcacheFile : FileMetadata marshalling fail, file: %s", fileName)
 		return nil, err
 	}
 	err = mm.CreateFileInit(fileName, fileMetadataBytes)
 	if err != nil {
-		log.Debug("DistributedCache::NewDcacheFile : File Creation failed for file :  %s with err : %s", fileName, err.Error())
+		log.Err("DistributedCache::NewDcacheFile : File Creation failed for file :  %s with err : %s", fileName, err.Error())
+		return nil, err
+	}
+	return &DcacheFile{
+		FileMetadata: fileMetadata,
+	}, nil
+}
+
+// Does all init process for opening the file.
+func OpenDcacheFile(fileName string) (*DcacheFile, error) {
+	fileMetadata, err := mm.GetFile(fileName)
+	if err != nil {
 		return nil, err
 	}
```

Comment:
in the else block we should check if the state is not "ready" then we should fail with an error that signifies noent to the caller, so that we can fail the fuse open call with ENOENT